config/initializers/devise.rb

```diff
@@ -178,13 +178,14 @@
 
   # ==> Configuration for :validatable
   # Range for password length.
-  config.password_length = 6..128
+  config.password_length = 8..128
 
   # Email regex used to validate email formats. It simply asserts that
   # one (and only one) @ exists in the given string. This is mainly
   # to give user feedback and not to assert the e-mail validity.
-  config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
-
+  # config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
+  config.email_regexp = /\A^([\w.-]+)@(\[(\d{1,3}\.){3}|(?!hotmail|gmail|googlemail|yahoo|gmx|ymail|outlook|bluewin|protonmail|t\-online|web\.|online\.|aol\.|live\.)(([a-zA-Z\d-]+\.)+))([a-zA-Z]{2,4}|\d{1,3})(\]?)$\z/
```

Comment:
Talvez não seja bacana editar as configurações direto no devise ( eu diria que seria difícil de buscar ) uma validação customizada com  menos dominíos restritos acho que atenderia. 
O pessoal da iugu-lite realizou algo assim em : `validators`
```
class CustomerEmailValidator < ActiveModel::Validator
  FORBIDDEN_DOMAINS = ['@iugu-lite.com.br',      # used for admins only
                       '@gmail.com', '@yahoo.com.br', '@hotmail.com.br'].freeze

  def validate(record)
    return unless record.email.end_with?(*FORBIDDEN_DOMAINS)

    record.errors.add(:email, I18n.t('.errors.forbidden_domain'))
  end
end
```
Fica legível e fácil de encontrar se for adicionar mais restrições seria só jogar no array o/

Reply:
@JorgeLAB, conversamos com o Henrique e para o nosso caso de uso com Admin e User é mais viável utilizar a validação no model do User, conforme a documentação:
https://guides.rubyonrails.org/active_record_validations.html#format